ompi/runtime/mpiruntime.h

```diff
@@ -55,6 +55,8 @@ OMPI_DECLSPEC extern bool ompi_mpi_initialized;
 OMPI_DECLSPEC extern bool ompi_mpi_finalized;
 /** Has the RTE been initialized? */
 OMPI_DECLSPEC extern bool ompi_rte_initialized;
+/** Did mpi start to finalize? */
+OMPI_DECLSPEC extern int32_t ompi_mpi_finalize_started;
```

Comment:
* similar functionality existed before this patch and "mpi_has_already_started" variable was local and int32_t. (see original before patch)

* the patch just externalized it, w/o changing logic or data layout.


``` shell
diff --git a/ompi/runtime/ompi_mpi_finalize.c b/ompi/runtime/ompi_mpi_finalize.c
index 51274992..7935ebeb 100644
--- a/ompi/runtime/ompi_mpi_finalize.c
+++ b/ompi/runtime/ompi_mpi_finalize.c
@@ -93,7 +93,6 @@ extern bool ompi_enable_timing_ext;
 int ompi_mpi_finalize(void)
 {
     int ret;
-    static int32_t finalize_has_already_started = 0;
     opal_list_item_t *item;
     ompi_proc_t** procs;
     size_t nprocs;
@@ -106,7 +105,7 @@ int ompi_mpi_finalize(void)
        ompi_comm_free() (or run into other nasty lions, tigers, or
        bears) */

-    if (! opal_atomic_cmpset_32(&finalize_has_already_started, 0, 1)) {
``` 

* It probably int32 because this variable was used for refcnt and allowed multiple invocation of finalize.